File: leetcode/762. Prime Number of Set Bits in Binary Representation/762.cpp
// https://helloacm.com/how-to-count-the-prime-number-of-set-bits-in-binary-representation/
// https://leetcode.com/problems/prime-number-of-set-bits-in-binary-representation/

class Solution {
public:
    int countPrimeSetBits(int L, int R) {
        int ans = 0;
        for (int x = L; x <= R; ++ x) {
            int bc = bitset<32>(x).count();
            if (bc == 2 || bc == 3 || bc == 5 || bc == 7 ||
                bc == 11 || bc == 13 || bc == 17 || bc == 19) ans ++;
        }
        return ans;        
    }
};

File: leetcode/762. Prime Number of Set Bits in Binary Representation/762.py
# https://helloacm.com/how-to-count-the-prime-number-of-set-bits-in-binary-representation/
# https://leetcode.com/problems/prime-number-of-set-bits-in-binary-representation/

class Solution:
    def countPrimeSetBits(self, L: int, R: int) -> int:
        ans = 0
        primes = [2, 3, 5, 7, 11, 13, 17, 19]
        for i in range(L, R + 1):
            bc = bin(i).count('1')
            if bc in primes:
                ans += 1
        return ans
